recipes/007_benchmarks_advanced.py

```diff
@@ -0,0 +1,66 @@
+"""
+.. _benchmarks_advanced:
+
+How to Implement Your Benchmark Problems with OptunaHub (Advanced)
+==================================================================
+
+OptunaHub provides the ``optunahub.benchmarks`` module for implementing benchmark problems.
+In this tutorial, we will explain how to implement complex benchmark problems such as a problem with dynamic search space using ``optunahub.benchmarks``.
+
+For the implementation of simple benchmark problems, please refer to :ref:`benchmarks_basic`.
+"""
+
+###################################################################################################
+# Implementing a Problem with Dynamic Search Space
+# -------------------------------------------------
+#
+# Here, let's implement a problem with a dynamic search space.
+#
+# First of all, import `optuna` and other required modules.
+from __future__ import annotations
+
+import optuna
+from optunahub.benchmarks import BaseProblem
+
+
+###################################################################################################
+# Next, define your own problem class by inheriting ``BaseProblem`` class.
+# To implement a problem with a dynamic search space, you should override the ``__call__(optuna.Trial)`` method that allows you to define the search space in a define-by-run manner.
```

Comment:
```suggestion
# To implement a problem with a dynamic search space, ``__call__(optuna.Trial)`` method must be overridden so that we can define a dynamic search space in the define-by-run manner.
```